driver_darwin.go

```diff
@@ -215,7 +215,7 @@ func oto_render(inUserData unsafe.Pointer, inAQ C.AudioQueueRef, inBuffer C.Audi
 		}
 	}
 
-	for i := 0; i < queueBufferSize; i++ {
+	for i := 0; i < len(buf); i++ {
 		*(*byte)(unsafe.Pointer(uintptr(inBuffer.mAudioData) + uintptr(i))) = buf[i]
```

Comment:
If `queueBufferSize < len(buf)`, should we fill the rest buffer with zeros?

Reply:
It may only appeared rarely after the closing called. At this moment it will dropped in the enqueue function since the context cancel has already called, and a mutex guaranteed it.

in my understanding, we may just simply return it when the len < size ,and a few other conditions. but it is not in the locked zoom. I am afraid always check all those variables may not efficient and may make the code hard to manage .


I may also add those conditions as well.

Reply:
I don't think just filling zeros is inefficient like this:

```
for i := len(buf); i < queueBufferSize; i++ {
    *(*byte)(unsafe.Pointer(uintptr(inBuffer.mAudioData) + uintptr(i))) = 0
}
```

In most cases this for loops is not executed as `len(buf) == queueBufferSize`.

> it is not in the locked zoom.

I don't understand this phrase.

Reply:
The root cause of their length are different is when it is continuing to prepare the buffer, and suddenly noticed the cancel had already called and ctx.Err() is not nil. And then it will jump out directly. 

It will only happened after the Close had already called and the driver.audioQueue is marking as nil. In this case, this whole inBuffer is mean less, and the final target  enqueueBuffer may do nothing. 

In this case, inBuffer is not important since it may never processed. 
The only reason of I didn’t add a lot of environment checking is I am hoping the code keep simple, and the cost is not too much.

Reply:
Thanks, then please leave a comment why we don't have to fill the buffer.

Reply:
ok, added